ccmbioinfo/lfs
# CARDS-1236: Test runMode should provide a mock NCR service which will allow the NCRNote component to be tested with hard-coded text annotations

This PR creates a servlet at `/ncr/annotate` which provides a mock NeuralCR service so that the NCRNote front-end component can be tested with hard-coded text annotations thus not requiring the CARDS test instance to be run in a Docker Compose environment that contains a NeuralCR container.

To test:
1. Build this branch
2. Start CARDS with the `test` runMode enabled
3. Install the HP vocabulary from BioPortal
4. Create a new _Concept Recognition Test_ Form
5. In the _Notes_ section for the _Co-morbidities_ question, the following texts (ensure that they are entered _exactly_ as shown) should be annotated as expected based on the hard-coded annotations provided by this PR:
  - `The patient has heart disease and diabetes`
  - `The patient has renal cancer and myopia`
  - `The patient has experienced chest pain and shortness of breath`
 
Please note that this mock NCR service will be used for all NCR annotations whenever the `test` runMode is enabled and a reverse proxy is not being used. This should not be a concern as in production environments the `test` runMode will not be enabled and a reverse proxy will be present to route everything under `/ncr/` to a real NCR container.
